pyccl/halos/halo_model.py

```diff
@@ -161,14 +147,23 @@ def profile_norm(self, cosmo, a, prof):
         Returns:
             float or array_like: integral value.
         """
-        # Compute mass function
-        self._get_ingredients(cosmo, a, False)
-        uk0 = prof.fourier(cosmo, self._prec['k_min'],
-                           self._mass, a, mass_def=self._mdef).T
-        norm = 1. / self._integrate_over_mf(uk0)
-        return norm
-
-    def number_counts(self, cosmo, sel, na=128, amin=None, amax=1.0):
+        self._get_ingredients(cosmo, a, get_bf=False)
+        uk0 = prof.fourier(cosmo, self.precision['k_norm'],
+                           self._mass, a, mass_def=self.mass_def).T
+        return 1 / self._integrate_over_mf(uk0)
```

Comment:
self-note: check this

Reply:
Is this about the `1 / 0` thing that throws these two warnings when we correlate GNFW at small redshifts?
If yes, we can probably leave it as is - I tested smaller numbers up to 1e-16 for `k_norm` but they don't change the fourier profile at all.

I eliminated that warning from tests using numpy's context manager `numpy.errstate`.
If you have a better implementation in mind, we can try something else.

Reply:
See my global comment about normprof. Will resolve this for now.